src/main/java/com/telenordigital/sms/smpp/pdu/DeliverSm.java

```diff
@@ -106,6 +108,18 @@ public static DeliverSm deserialize(final ByteBuf buf, final Charset defaultChar
         opts.getByteArray(TlvTag.NETWORK_ERROR_CODE)
             .map(nc -> "0x%06x".formatted(new BigInteger(1, nc)));
 
+    final var mcc =
```

Comment:
I wonder if we rather create a record type containing mcc and mnc?  Something like SubAddress(int mcc, int mnc) {}

Reply:
good idea